### ISSUE TITLE
Problem with \fillin when at start of line

### ISSUE THREAD

**Issue by mitchkeller** [PreTeXtBook/pretext#929]
Preview Activity 1.2.1 in Active Calculus is demonstrating a small issue with @Alex-Jordan's LaTeX code for `\fillin`. With the way the sentence containing `<fillin>` is currently phrased, the blank winds up at the start of a line. However, LaTeX fails to actually produce the line in this case. If I insert some extra words before the `\fillin`, then the line is drawn in the output. If I manually insert a `\newline` into the LaTeX before `\fillin`, the line still isn't rendered (and no horizontal space is consumed—the period and close quote appear at the start of the line).

Resolving this is well beyond my level of TeXpertise, unfortunately. I'm going to hack something that gets the blank to appear in the 2018 edition of Active Calculus, and then this can get fixed later.

**Comment by Alex-Jordan** [comment]
Hmm, sorry about this. I suggest for now that you add something to `latex.preamble.late` that renews `\fillin`. The code right now looks so complicated so that it can support the box style, and also so that it can be very stretchy/compressible. If it didn't have to support the box style, it could be a lot simpler. The XSLT could change to make the definition a lot simpler in the underscored case. That would make a bigger if/then in the XSLT, but a cleaner .tex when using underlined fillins.

Or maybe there is a similar problem if using the box style. 

Here is another thing you could try in the short term. Note how the code allows the fillin to compress all the way to 0 width. Edit so that something like 1em is minimal and see what effect that has. Of course, it might make answer blanks move on to new lines where previously there was a tiny answer blank on the previous line.

**Comment by mitchkeller** [comment]
The definition of `\fillin` that I'm getting now is

```
\newcommand{\fillin}[1]{\leavevmode\leaders\vrule height -1.2pt depth 1.5pt \hskip #1em minus #1em \null}
```

I'm really bad at rule drawing commands in (La)TeX, so I really can't make heads or tails out of this. I'm not sure how to change the minimum width in this command.

**Comment by Alex-Jordan** [comment]
There is a place where you see `\hskip`. It takes the argument `#1` times as many em, but adds rubber space of negative that many em. So it starts wanting to go `#1em` forward, but is willing to shrink back down to 0 if that helps fit something nicely on a line. Not sure if you can get away with `\hskip #1em minus (#1-1)em`. Maybe you need to use `\dimexpr` like 
```
\hskip #1em minus \dimexpr#1em-1em\relax`

Sorry I can'ttest right now, but that is the basic logic.
```

**Comment by mitchkeller** [comment]
That code also results in a nonexistent (probably length 0) line. I've just pulled the old `\fillin` macro that was used for the 2017 edition for inspiration and am using `\underline{\hspace{4.5em}}` for the one problematic `\fillin`. I've got the rest of the book ready to go to CreateSpace, so prefer not to muck about with things that could have more global repercussions right now. Happy to help debug later if needed.
